### PR TITLE
fix(analyze-overview): Require user to be logged in to add to space

### PR DESCRIPTION
**Pull request by c-koehler**
fixes https://github.com/openshiftio/openshift.io/issues/868

### CODE REVIEW

**Comment by fabric8cd** [comment]
@c-koehler fabric8/fabric8-ui:SNAPSHOT-PR-2274-1 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2274-fabric8-ui.badger.fabric8.io

File: src/app/space/analyze/analyze-overview/analyze-overview.component.ts [hunk 1]
```diff
@@ -36,8 +45,12 @@ export class AnalyzeOverviewComponent implements OnInit, OnDestroy {
   }
 
   openForgeWizard(addSpace: TemplateRef<any>) {
-    this.selectedFlow = 'selectFlow';
-    this.modalRef = this.modalService.show(addSpace, {class: 'modal-lg'});
+    if (this.authService.isLoggedIn()) {
```
Comment (dgutride) [inline src/app/space/analyze/analyze-overview/analyze-overview.component.ts]: @catrobson - this seems to imply that the user sees the add to space button when they aren't authorized to do anything, should we be hiding the button instead?

Comment (catrobson) [reply src/app/space/analyze/analyze-overview/analyze-overview.component.ts]: If the user is not logged in, the button shouldn't be shown to them.  I'm not sure what "authorized" means in this context - is that something more than logged in/logged out?

Comment (c-koehler) [reply src/app/space/analyze/analyze-overview/analyze-overview.component.ts]: The box has been hidden if there is no logged in user, however there is still logic there that will prevent a user from trying to do this action if somehow there is another way to access the code that the button invokes.

Do we want to still leave that code there as a safeguard, or should it be removed?

Comment (c-koehler) [reply src/app/space/analyze/analyze-overview/analyze-overview.component.ts]: @dgutride Checking in for any input above (sorry, I probably should have highlighted in my post)

**Comment by c-koehler** [comment]
@dgutride Checking in again for any comments on the above

File: src/app/space/analyze/analyze-overview/analyze-overview.component.ts (hunk 1 above)
Comment (dgutride) [reply src/app/space/analyze/analyze-overview/analyze-overview.component.ts]: I'd remove that logic from launching the wizard, I have another PR coming which is doing something similar related to the github linking and provides feedback if they aren't linked.

Comment (c-koehler) [reply src/app/space/analyze/analyze-overview/analyze-overview.component.ts]: @dgutride To confirm: should the code changes to this file be reset back to being untouched, and then I revisit this later when your PR gets in (and then use what you have added)? Or does your PR take care of everything and I should just remove what I did here and leave it?

**Comment by joshuawilson** [comment]
please rebase

File: src/app/space/analyze/analyze-overview/analyze-overview.component.ts (hunk 1 above)
Comment (dgutride) [reply src/app/space/analyze/analyze-overview/analyze-overview.component.ts]: Your PR will be useful for those not logged in at all.  Hiding the button will still be necessary - my PR will handle what happens if somebody launches the wizard (logged in but missing some permissions)

File: src/app/space/analyze/analyze-overview/analyze-overview.component.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
       <fabric8-edit-space-description-widget></fabric8-edit-space-description-widget>
     </div>
     <div class="col-xs-4 col-xs-offset-4 col-sm-2 col-sm-offset-0">
-      <button id="analyze-overview-add-to-space-button" class="btn btn-primary btn-lg pull-right" (click)="openForgeWizard(addSpace)">Add to space</button>
+      <button *ngIf="this.authService.isLoggedIn()" id="analyze-overview-add-to-space-button" class="btn btn-primary btn-lg pull-right" (click)="openForgeWizard(addSpace)">Add to space</button>
```
Comment (dgutride) [inline src/app/space/analyze/analyze-overview/analyze-overview.component.html]: This change is good - please keep it.

File: src/app/space/analyze/analyze-overview/analyze-overview.component.ts (hunk 1 above)
Comment (dgutride) [inline src/app/space/analyze/analyze-overview/analyze-overview.component.ts]: There is no need to check on wizard launching now - that is covered by the github connect PR I have.

**Comment by c-koehler** [comment]
@dgutride Updated to use only the button visibility commit

**Comment by fabric8cd** [comment]
@c-koehler fabric8/fabric8-ui:SNAPSHOT-PR-2274-4 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2274-fabric8-ui.badger.fabric8.io

**Pull request merged**